searx/poolrequests.py

```diff
@@ -178,24 +357,23 @@ def request(method, url, **kwargs):
         del kwargs['raise_for_httperror']
 
     # do request
-    response = session.request(method=method, url=url, **kwargs)
-
-    time_after_request = time()
-
-    # is there a timeout for this engine ?
-    if timeout is not None:
-        timeout_overhead = 0.2  # seconds
-        # start_time = when the user request started
-        start_time = getattr(threadLocal, 'start_time', time_before_request)
-        search_duration = time_after_request - start_time
-        if search_duration > timeout + timeout_overhead:
-            raise requests.exceptions.Timeout(response=response)
-
-    # session end
-    session.close()
-
-    if hasattr(threadLocal, 'total_time'):
-        threadLocal.total_time += time_after_request - time_before_request
+    future = asyncio.run_coroutine_threadsafe(send_request(method, url, get_enable_http_protocol(), kwargs), LOOP)
+    try:
+        if timeout:
```

Comment:
Please move out lines that cannot raise the expectation you want to handle in this block.

Reply:
Done in the last commit.